cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/SchemaService.java

```diff
@@ -118,12 +107,31 @@ public synchronized ApiResponse registerSchema(ClusterSchemaRequest clusterSchem
       return ApiResponse.builder().result("Failure in registering schema.").build();
     } finally {
       // Ensure the Schema compatibility is returned to previous setting before the force update.
-      // (Set isForce to false in all cases to revert it.
-      if (clusterSchemaRequest.isForceRegister() && schemaCompatibilityCompleted) {
-        log.info(
-            "RegisterSchema - Force Commit revert to original Schema Compatibility {} for Topic {}",
-            schemaCompatibility,
-            clusterSchemaRequest.getTopicName());
+      resetCompatibilityOnSubject(
+          clusterSchemaRequest, schemaCompatibility, schemaCompatibilitySetOnSubject);
```

Comment:
~if it's not "force register" `schemaCompatibility` will be null. I don't think we want this.
As I understand, this action should only be performed if `schemaCompatibility` is not null (also would cover the case that an exception is thrown when retrieving its value).~
Nevermind, the function internals do these checks.